Sources/Optimizely/OptimizelyClient.swift

```diff
@@ -771,12 +775,14 @@ extension OptimizelyClient {
             // send notification in sync mode (functionally same as async here since it's already in background thread),
             // but this will make testing simpler (timing control)
 
-            self.sendActivateNotification(experiment: experiment,
-                                          variation: variation,
-                                          userId: userId,
-                                          attributes: attributes,
-                                          event: event,
-                                          async: false)
+            if let tmpVariation = variation {
```

Comment:
Why need to add this condition?

Reply:
Variation was non-optional before so we didn't have to unwrap it, but since we have allowed impression event's to be sent with nil variation, this is necessary!